components/src/TimePicker/TimePicker.js

```diff
@@ -96,43 +119,139 @@ const TimePicker = ({
   maxTime,
   defaultValue,
   onInputChange,
-  "aria-label": ariaLabel,
-  ...props
+  onBlur,
+  onFocus,
+  errorMessage,
+  errorList,
+  labelText,
+  placeholder,
+  onClick,
+  onChange,
+  "aria-label": ariaLabel
 }) => {
-  const [input, setInput] = useState("");
+  const [input, setInput] = useState(defaultValue);
+  const [currentOptionRef, setCurrentOptionRef] = useState(null);
+  const [dropdownIsOpen, setDropdownIsOpen] = useState(false);
   const { locale } = useContext(LocaleContext);
+  const [ref, setRef] = useState(null);
   const { t } = useTranslation();
 
+  useEffect(() => {
+    if (currentOptionRef && dropdownIsOpen) {
+      currentOptionRef.scrollIntoView({
+        behaviour: "smooth",
+        block: "center"
+      });
+    }
+  }, [currentOptionRef, dropdownIsOpen]);
+
   const filteredOptions = () => {
-    const inputHasMinutes = /[:][0-9]/.test(standardizeTime(input));
     const optionsAtInterval = getTimeOptions(interval, timeFormat, minTime, maxTime, locale) || [];
-    const optionsByMinute = getTimeOptions(1, timeFormat, minTime, maxTime, locale) || [];
-    const optionsList = inputHasMinutes ? optionsByMinute : optionsAtInterval;
-    return optionsList.filter(({ label }) => standardizeTime(label).includes(standardizeTime(input)));
+    return optionsAtInterval;
   };
 
-  const overrideInternalFiltering = () => true;
   const visibleOptions = filteredOptions() || [];
 
+  const hasError = !!(errorMessage || errorList);
+
+  const handleBlur = e => {
+    onBlur(e);
+    setDropdownIsOpen(false);
+    const optionsByMinute = getTimeOptions(1, timeFormat, minTime, maxTime, locale);
+    if (optionsByMinute.filter) {
+      setInput(optionsByMinute.filter(({ label }) => standardizeTime(label).includes(standardizeTime(input)))[0].label);
+    }
+  };
+
+  const handleFocus = e => {
+    onFocus(e);
+  };
+
+  const handleClickInput = e => {
+    onClick(e);
+    setDropdownIsOpen(true);
+  };
+
+  const handleOptionSelection = option => {
+    setInput(option.label);
+    setDropdownIsOpen(false);
+    onChange(option);
+  };
+
+  const isClosestTime = ({ index }) => {
+    return getIntervalFromTime(input, interval, minTime) === index;
+  };
+
+  const onCurrentOptionRefChange = React.useCallback(node => {
+    if (node) {
+      setCurrentOptionRef(node);
+    }
+  }, []);
+
+  const onRefChange = React.useCallback(node => {
+    if (node) {
+      setRef(node);
+    }
+  }, []);
+
+  const handleKeyDown = (event, option) => {
+    if (event.keyCode === 40) {
+      // key down
+      setDropdownIsOpen(true);
+    } else if (event.keyCode === 13) {
```

Comment:
I thiiink we have a file that stores all our keycodes as constants so you don't have to comment like this.